gotooltest/setup.go

```diff
@@ -36,12 +37,22 @@ var (
 
 // initGoEnv initialises goEnv. It should only be called using goEnv.once.Do,
 // as in Setup.
+//
+// Run all of these probe commands in a temporary directory, so as not to make
+// any assumptions about the caller's working directory.
 func initGoEnv() error {
 	var err error
 
+	td, err := os.MkdirTemp("", "gotooltest-initGoEnv")
+	if err != nil {
+		return fmt.Errorf("failed to create temporary directory for go command tests: %w", err)
+	}
+	defer os.RemoveAll(td)
```

Comment:
nit: `os.Remove` is enough given that you know it is an empty directory

also: this ignores errors, which could mean silently leaking temporary dirs/files. perhaps do a defer which still captures or logs the error?

Reply:
I've left `os.RemoveAll()` because I now write a temporary `go.mod` file to prevent parent directories' `go.mod` files interfering in any way.

Reply:
The error is now captured in the case that the function otherwise returns no error.